Galois cannot be imported when NumPy 2.4.0 is installed

NumPy 2.4.0 removed `np.trapz`:
https://numpy.org/doc/stable/release/2.4.0-notes.html#removal-of-deprecated-functions-and-arguments

This causes Galois to fail to import with this error:
````
../test_dir/usr/lib/python3.13/site-packages/galois/__init__.py:22: in <module>
    from ._domains import *
../test_dir/usr/lib/python3.13/site-packages/galois/_domains/__init__.py:5: in <module>
    from ._array import *
../test_dir/usr/lib/python3.13/site-packages/galois/_domains/_array.py:18: in <module>
    from ._function import FunctionMixin
../test_dir/usr/lib/python3.13/site-packages/galois/_domains/_function.py:300: in <module>
    class FunctionMixin(np.ndarray, metaclass=ArrayMeta):
../test_dir/usr/lib/python3.13/site-packages/galois/_domains/_function.py:314: in FunctionMixin
    np.trapz,
    ^^^^^^^^
/usr/lib/python3.13/site-packages/numpy/__init__.py:792: in __getattr__
    raise AttributeError(f"module {__name__!r} has no attribute {attr!r}")
E   AttributeError: module 'numpy' has no attribute 'trapz'. Did you mean: 'trace'?
````

Sure, I don't think it will cause any harm. But it also doesn't solve the issue in Arch Linux :)

```
    import galois
../test_dir/usr/lib/python3.13/site-packages/galois/__init__.py:24: in <module>
    from ._fields import *
../test_dir/usr/lib/python3.13/site-packages/galois/_fields/__init__.py:6: in <module>
    from ._factory import *
../test_dir/usr/lib/python3.13/site-packages/galois/_fields/_factory.py:21: in <module>
    from ._ufunc import UFuncMixin_2_m, UFuncMixin_p_1, UFuncMixin_p_m
../test_dir/usr/lib/python3.13/site-packages/galois/_fields/_ufunc.py:9: in <module>
    from .._domains import _calculate, _lookup
../test_dir/usr/lib/python3.13/site-packages/galois/_domains/_calculate.py:22: in <module>
    @numba.jit(["int64[:](int64, int64, int64)"], nopython=True, cache=True)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.13/site-packages/numba/core/decorators.py:232: in wrapper
    disp.compile(sig)
/usr/lib/python3.13/site-packages/numba/core/dispatcher.py:885: in compile
    cres = self._cache.load_overload(sig, self.targetctx)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.13/site-packages/numba/core/caching.py:679: in load_overload
    target_context.refresh()
/usr/lib/python3.13/site-packages/numba/core/base.py:267: in refresh
    self.load_additional_registries()
/usr/lib/python3.13/site-packages/numba/core/cpu.py:76: in load_additional_registries
    from numba.np import linalg, arraymath, arrayobj # noqa F401
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.13/site-packages/numba/np/arraymath.py:6: in <module>
    sys.modules[__name__] = _RedirectSubpackage(locals(),
/usr/lib/python3.13/site-packages/numba/core/utils.py:694: in __init__
    new_mod_obj = import_module(new_module)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^
/usr/lib/python3.13/site-packages/numba/np/old_arraymath.py:2218: in <module>
    @overload(np.trapz)
              ^^^^^^^^
/usr/lib/python3.13/site-packages/numpy/__init__.py:792: in __getattr__
    raise AttributeError(f"module {__name__!r} has no attribute {attr!r}")
E   AttributeError: module 'numpy' has no attribute 'trapz'. Did you mean: 'trace'?
```
I will need to patch Numba or wait for the new version.

--- a/src/galois/_domains/_function.py
+++ b/src/galois/_domains/_function.py
@@ -311,7 +311,6 @@ class FunctionMixin(np.ndarray, metaclass=ArrayMeta):
         np.round,
         np.fix,
         np.gradient,
-        np.trapz,
         np.i0,
         np.sinc,
         np.angle,
@@ -323,6 +322,11 @@ class FunctionMixin(np.ndarray, metaclass=ArrayMeta):
         np.lib.scimath.logn,
         np.cross,
     ]
+
+    if np.lib.NumpyVersion(np.__version__) < "2.4.0":
+        _UNSUPPORTED_FUNCTIONS.append(np.trapz)
+    else:
+        _UNSUPPORTED_FUNCTIONS.append(np.trapezoid)
 
     _FUNCTIONS_REQUIRING_VIEW = [
         np.concatenate,

